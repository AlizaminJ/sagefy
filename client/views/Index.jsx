import React from 'react'
import { StaticRouter, Route, Switch } from 'react-router-dom'
import { string } from 'prop-types'

import HomePage from './pages/HomePage'
import ContactPage from './pages/ContactPage'
import TermsPage from './pages/TermsPage'
import NotFoundPage from './pages/NotFoundPage'
import SearchSubjectsPage from './pages/SearchSubjectsPage'

export const output = a => () => a
const searchSubjects = output('Search Subjects')
const contact = output('Contact')
const terms = output('Terms')
const home = output('Home')
const notFound = output('Not Found')

export default function Index({ location }) {
  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1.0"
        />
        <link rel="stylesheet" href="/sagefy.min.css" />
        <title>
          <StaticRouter context={{}} location={location}>
            <Switch>
              <Route path="/search-subjects" render={searchSubjects} />
              <Route path="/contact" render={contact} />
              <Route path="/terms" render={terms} />
              <Route path="/" exact render={home} />
              <Route render={notFound} />
            </Switch>
          </StaticRouter>{' '}
          â€“ Sagefy
        </title>
      </head>
      <body>
        <div id="top" className="page" role="document">
          <StaticRouter context={{}} location={location}>
            <Switch>
              <Route path="/search-subjects" component={SearchSubjectsPage} />
              <Route path="/contact" component={ContactPage} />
              <Route path="/terms" component={TermsPage} />
              <Route path="/" exact component={HomePage} />
              <Route component={NotFoundPage} />
            </Switch>
          </StaticRouter>
        </div>
      </body>
    </html>
  )
}

Index.propTypes = {
  location: string.isRequired,
}
