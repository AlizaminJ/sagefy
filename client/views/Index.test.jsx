import Index from './Index'

describe('Index view', () => {
  it('should render a page', () => {
    expect(Index({ location: '/' })).toMatchSnapshot()
  })
})
