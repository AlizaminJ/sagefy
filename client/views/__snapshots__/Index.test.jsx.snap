// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index view should render a page 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Switch>
    <Route
      path="/subjects/:subjectId"
      render={[Function]}
    />
    <Route
      path="/choose-step"
      render={[Function]}
    />
    <Route
      path="/learn-video/:cardId"
      render={[Function]}
    />
    <Route
      path="/learn-page/:cardId"
      render={[Function]}
    />
    <Route
      path="/learn-unscored-embed/:cardId"
      render={[Function]}
    />
    <Route
      path="/learn-choice/:cardId"
      render={[Function]}
    />
    <Route
      path="/log-in"
      render={[Function]}
    />
    <Route
      path="/password"
      render={[Function]}
    />
    <Route
      path="/email"
      render={[Function]}
    />
    <Route
      path="/settings"
      render={[Function]}
    />
    <Route
      path="/dashboard"
      render={[Function]}
    />
    <Route
      path="/sign-up"
      render={[Function]}
    />
    <Route
      path="/search-subjects"
      render={[Function]}
    />
    <Route
      path="/create-subject"
      render={[Function]}
    />
    <Route
      path="/create-card"
      render={[Function]}
    />
    <Route
      path="/create-choice-card"
      render={[Function]}
    />
    <Route
      path="/create-page-card"
      render={[Function]}
    />
    <Route
      path="/create-video-card"
      render={[Function]}
    />
    <Route
      path="/create-unscored-embed-card"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/contact"
    />
    <Route
      component={[Function]}
      path="/terms"
    />
    <Route
      component={[Function]}
      path="/server-error"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</StaticRouter>
`;
